session/session.go

```diff
@@ -98,8 +98,10 @@ func newZKSession(servers string, recvTimeout time.Duration, logger stdLogger, c
 	var event zookeeper.Event
 	select {
 	case event = <-events:
-		if event.State != zookeeper.STATE_CONNECTED {
+		if event.State == zookeeper.STATE_AUTH_FAILED || event.State == zookeeper.STATE_EXPIRED_SESSION {
 			return nil, ErrZKSessionNotConnected
+		} else {
+			break
```

Comment:
@csfrancis if there is an event in the events channel and it isnt `STATE_EXPIRED_SESSION` or `STATE_AUTH_FAILED` then it is connected. no?